template.yaml

```diff
@@ -388,14 +397,57 @@ Resources:
 
               # Get dynamically the node name when launching slurmd
               cat > $SLURM_HOME/etc/aws/get_nodename <<'EOF'
-              instanceid=`/usr/bin/curl --fail -m 2 -s http://169.254.169.254/latest/meta-data/instance-id`
-              if [[ ! -z "$instanceid" ]]; then
-                 hostname=`/usr/bin/curl -s http://169.254.169.254/latest/meta-data/tags/instance/Name`
+              #!/bin/bash
+
+              # Function to get IMDSv2 token
```

Comment:
If you switch to IMDSv2, then the instance creation should be set to IMDSv2 and avoid rely on default.